### ISSUE TITLE
During the learning I have a crash

### ISSUE THREAD

**Issue by Franco7Scala** [dubezOniner/Deep-QLearning-Demo-csharp#1]
This is the neural network:

`int quantityShootingInputs = 8;
        int quantityShootingActions = 4;
        int networkShootingSize = quantityMovementInputs * temporalWindow + quantityMovementActions * temporalWindow + quantityMovementInputs;
        List<LayerDefinition> shootingLayers = new List<LayerDefinition>();
        shootingLayers.Add(new LayerDefinition { type = "input", out_sx = 1, out_sy = 1, out_depth = networkShootingSize });
        shootingLayers.Add(new LayerDefinition { type = "fc", num_neurons = 70, activation = "relu" });
        shootingLayers.Add(new LayerDefinition { type = "fc", num_neurons = 70, activation = "relu" });
        shootingLayers.Add(new LayerDefinition { type = "fc", num_neurons = 70, activation = "relu" });
        shootingLayers.Add(new LayerDefinition { type = "regression", num_neurons = quantityShootingActions });
// training configuration
        TrainingOptions trainingOptions = new TrainingOptions();
        trainingOptions.temporal_window = temporalWindow;
        trainingOptions.experience_size = 30000;
        trainingOptions.start_learn_threshold = 1000;
        trainingOptions.gamma = 0.7;
        trainingOptions.learning_steps_total = 100000;
        trainingOptions.learning_steps_burnin = 3000;
        trainingOptions.epsilon_min = 0.05;
        trainingOptions.epsilon_test_time = 0.00;
        trainingOptions.layer_defs = shootingLayers;
        trainingOptions.options = new Options { method = "adadelta", l2_decay = 0.001, batch_size = 10 };
        neuralNetwork = new DeepQLearn(quantityInputs, quantityActions, trainingOptions);`

and this is the error:

`IndexOutOfRangeException: Array index is out of range.
ConvnetSharp.FullyConnectedLayer.forward (ConvnetSharp.Volume V, Boolean is_training) (at Assets/Scripts/Player/NeuralEngines/BySensors/Layers/FullyConnectedLayer.cs:47)
ConvnetSharp.Net.forward (ConvnetSharp.Volume V, Boolean is_training) (at Assets/Scripts/Player/NeuralEngines/BySensors/Utilities/Net.cs:130)
DeepQLearning.DRLAgent.DeepQLearn.policy (System.Double[] s) (at Assets/Scripts/Player/NeuralEngines/BySensors/Q-Learning/DeepQLearn.cs:241)
DeepQLearning.DRLAgent.DeepQLearn.backward (Double reward) (at Assets/Scripts/Player/NeuralEngines/BySensors/Q-Learning/DeepQLearn.cs:381)
SensorAgent.Backward (Double reward) (at Assets/Scripts/Player/NeuralEngines/BySensors/SensorAgent.cs:79)
NeuralBySensorPlayer.Update () (at Assets/Scripts/Player/NeuralBySensorPlayer.cs:115)
`

The crash there is when start the learning and if I decrease the temporal window at 1 it works!
